eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php

```diff
@@ -401,12 +403,36 @@ public function moveSubtreeNodes(array $sourceNodeData, array $destinationNodeDa
         }
     }
 
-    private function isHiddenByParent($pathString, array $rows)
+    private function getHiddenNodeIds(int $contentObjectId)
     {
-        $parentNodeIds = explode('/', trim($pathString, '/'));
+        $query = $this->buildHiddenSubtreeQuery('node_id');
+        $expr = $query->expr();
+        $query
+            ->andWhere(
+                $expr->eq(
+                    'id',
+                    $query->createPositionalParameter(
+                        $contentObjectId,
+                        PDO::PARAM_INT
```

Comment:
Using PDO consts is deprecated
```suggestion
                        ParameterType::INTEGER
```
\+
```php
use Doctrine\DBAL\ParameterType;
```

Reply:
This is actually one thing I changed when copying from ezplatform-kernel. That was done in order to be consistent. In this file `PDO::PARAM_*` is already used. You want me to change to `ParameterType::*` all-over in this class, or only on the lines this PR already touches?

Reply:
> This is actually one thing I changed when copying from ezplatform-kernel. That was done in order to be consistent. In this file `PDO::PARAM_*` is already used. You want me to change to `ParameterType::*` all-over in this class, or only on the lines this PR already touches?

Let's keep it simple - only affected methods.

Reply:
Ok, fixed in https://github.com/ezsystems/ezpublish-kernel/pull/3142/commits/66ee6a667394b4bb962d330baef104e4e07ef042